require can't find npm modules in node_modules dir

``` (sh)
~/Sites/test $ npm install winston
~/Sites/test $ echo "require('winston');console.log('ok')" > test.js
~/Sites/test $ traceur test.js

module.js:340
    throw err;
          ^
Error: Cannot find module 'winston'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (test.js:1:63)
    at Module._compile (module.js:456:26)
    at /usr/local/share/npm/lib/node_modules/traceur/src/node/interpreter.js:34:18
    at /usr/local/share/npm/lib/node_modules/traceur/src/node/inline-module.js:204:9
    at $CodeUnit.dispatch_ (eval at <anonymous> (/usr/local/share/npm/lib/node_modules/traceur/src/node/traceur.js:24:17), <anonymous>:18883:13)
    at $CodeUnit.dispatchComplete (eval at <anonymous> (/usr/local/share/npm/lib/node_modules/traceur/src/node/traceur.js:24:17), <anonymous>:18871:14)

~/Sites/test $ node test.js
ok
```

Seems require needs the full path to module.

This works:

```
var winston = require(process.cwd() + '/node_modules/winston');
```

--- a/src/node/interpreter.js
+++ b/src/node/interpreter.js
@@ -15,6 +15,7 @@
 'use strict';
 
 var fs = require('fs');
+var Module = require('module');
 var traceur = require('./traceur.js');
 var util = require('./file-util.js');
 var inlineAndCompile = require('./inline-module.js').inlineAndCompile;
@@ -22,20 +23,26 @@ var inlineAndCompile = require('./inline-module.js').inlineAndCompile;
 var ErrorReporter = traceur.util.ErrorReporter;
 var TreeWriter = traceur.outputgeneration.TreeWriter;
 
+var ext = '.traceur-compiled';
+
+Module._extensions[ext] = function(module, filename) {
+  module.filename = filename.slice(0, -ext.length);
+  module._compile(module.compiledCode, module.filename);
+};
+
 function interpret(filename, argv, flags) {
   var reporter = new ErrorReporter();
-  var mainModule = require.main;
-  var execArgv = [mainModule.filename].concat(flags || []);
+  var execArgv = [require.main.filename].concat(flags || []);
 
+  filename = fs.realpathSync(filename);
   process.argv = ['traceur', filename].concat(argv || []);
   process.execArgv = process.execArgv.concat(execArgv);
 
   inlineAndCompile([filename], {}, reporter, function(tree) {
-    var compiledCode = TreeWriter.write(tree);
+    var module = new Module(filename, require.main);
 
-    // TODO: use a new module rather than the main one.
-    mainModule.filename = fs.realpathSync(filename);
-    mainModule._compile(compiledCode, mainModule.filename);
+    module.compiledCode = TreeWriter.write(tree);
+    module.load(filename + ext);
   }, function(err) {
     console.error(err);
     process.exit(1);

